plugin/src/main/java/org/opensearch/ml/model/MLModelAutoReLoader.java

```diff
@@ -0,0 +1,271 @@
+        MLLoadModelRequest mlLoadModelRequest = new MLLoadModelRequest(modelId, new String[] { nodeId }, false, false);
+        return client.execute(MLLoadModelAction.INSTANCE, mlLoadModelRequest).actionGet(5000);
+    }
+
+    /**
+     * get retry times from the index ".plugins-ml-model-reload" by 1 ml node
+     * @param nodeId the filter condition to query
+     * @return retry times
+     */
+    @VisibleForTesting
+    int getReTryTimes(String nodeId) {
+        if (!isExistedIndex(ML_MODEL_RELOAD_INDEX)) {
+            return 0;
+        }
+
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+        searchSourceBuilder.fetchSource(new String[] { MODEL_LOAD_RETRY_TIMES_FIELD }, null);
+        QueryBuilder queryBuilder = QueryBuilders.idsQuery().addIds(nodeId);
+        searchSourceBuilder.query(queryBuilder);
+        SearchRequest searchRequest = new SearchRequest().source(searchSourceBuilder).indices(ML_MODEL_RELOAD_INDEX);
+
+        try {
+            SearchResponse response = client.execute(SearchAction.INSTANCE, searchRequest).actionGet(5000);
+
+            SearchHit[] hits = response.getHits().getHits();
+            if (CollectionUtils.isEmpty(hits)) {
+                return 0;
+            }
+
+            Map<String, Object> sourceAsMap = hits[0].getSourceAsMap();
+            return (Integer) sourceAsMap.get(MODEL_LOAD_RETRY_TIMES_FIELD);
+        } catch (IndexNotFoundException e) {
+            throw new IndexNotFoundException("index " + ML_MODEL_RELOAD_INDEX + " not found");
+        }
+    }
+
+    /**
+     * judge whether the index ".plugins-ml-model-reload" existed
+     * @param indexName index name
+     * @return true: existed. false: not existed
+     */
+    @VisibleForTesting
+    boolean isExistedIndex(String indexName) {
+        IndicesExistsRequest existsRequest = new IndicesExistsRequest(indexName);
+        IndicesExistsResponse exists = client.execute(IndicesExistsAction.INSTANCE, existsRequest).actionGet(5000);
+
+        return exists.isExists();
+    }
+
+    /**
+     * save retry times
+     * @param nodeId node id
+     * @param reTryTimes actual retry times
+     */
+    @VisibleForTesting
+    void saveLatestReTryTimes(String nodeId, int reTryTimes) {
+        Map<String, Object> content = new HashMap<>();
+        content.put(NODE_ID_FIELD, nodeId);
+        content.put(MODEL_LOAD_RETRY_TIMES_FIELD, reTryTimes);
+
+        IndexRequest indexRequest = new IndexRequest(ML_MODEL_RELOAD_INDEX);
+        indexRequest.id(nodeId);
+        indexRequest.source(content);
+        indexRequest.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+
+        IndexResponse indexResponse = client.execute(IndexAction.INSTANCE, indexRequest).actionGet(5000);
+
+        if (indexResponse.status() == RestStatus.CREATED || indexResponse.status() == RestStatus.OK) {
+            log.debug("node id:{} insert retry times successfully", nodeId);
```

Comment:
why using debug level here? Change it to info.

Reply:
except in try-catch, we just use debug to min the effort by log